keras_nlp/src/models/resnet/resnet_backbone.py

```diff
@@ -461,7 +527,295 @@ def apply_bottleneck_block(
             use_bias=False,
             dtype=dtype,
             name=f"{name}_0_conv",
+        )(shortcut)
```

Comment:
What's up with this 50 or so line addition to the basic block? Shouldn't all the changes be the VD blocks now? Am I missing something?

Reply:
Just as the normal ResNet, ResNet_vds use basic blocks in their small versions, and the bigger ResNet_vds use bottleneck blocks. Therefore, we need vd variants of both, `basic_block` and `bottleneck_block`.

Reply:
Oh I think this was just github diff being weird, my bad! It looked like you were making a massive addition to the non-vd block, but I see now that's just because the code matches across both. Let's try this out!